crates/integration_tests/Cargo.toml

```diff
@@ -27,9 +27,11 @@ rust-version = { workspace = true }
 [dependencies]
 arrow-array = { workspace = true }
 arrow-schema = { workspace = true }
+datafusion = "43"
```

Comment:
The version here differs from the one in the datafusion integration crate: https://github.com/apache/iceberg-rust/blob/105e513ac528d837c21e680fce2d210f3c1fb795/crates/integrations/datafusion/Cargo.toml#L34.

Can we add datafusion to the workspace level `Cargo.toml` and update both here and the datafusion integration crate to `workspace = true`?

Reply:
Makes sense, done!